query-engine/query-engine/src/main.rs

```diff
@@ -49,42 +41,37 @@ async fn main() -> Result<(), AnyError> {
 
     async fn main() -> Result<(), PrismaError> {
         let opts = PrismaOpt::from_args();
+        init_logger(&opts);
         feature_flags::initialize(opts.raw_feature_flags.as_slice())?;
         match CliCommand::from_opt(&opts)? {
             Some(cmd) => cmd.execute().await?,
             None => {
-                set_panic_hook();
+                set_panic_hook(&opts);
                 server::listen(opts).await?;
             }
         }
         Ok(())
     }
 }
 
-fn init_logger() -> Result<(), AnyError> {
-    match *LOG_FORMAT {
-        LogFormat::Text => {
-            let subscriber = FmtSubscriber::builder()
-                .with_env_filter(EnvFilter::from_default_env())
-                .finish();
+fn init_logger(opts: &PrismaOpt) {
+    // Enable `tide` logs to be captured.
+    let filter = EnvFilter::from_default_env().add_directive("tide=info".parse().unwrap());
 
-            subscriber::set_global_default(subscriber)?;
+    match opts.log_format() {
+        LogFormat::Text => {
+            let subscriber = FmtSubscriber::builder().with_env_filter(filter).finish();
+            subscriber::set_global_default(subscriber).expect("Could not initialize logger");
         }
         LogFormat::Json => {
-            let subscriber = FmtSubscriber::builder()
-                .json()
-                .with_env_filter(EnvFilter::from_default_env())
-                .finish();
-
-            subscriber::set_global_default(subscriber)?;
+            let subscriber = FmtSubscriber::builder().json().with_env_filter(filter).finish();
+            subscriber::set_global_default(subscriber).expect("Could not initialize logger");
         }
     }
-
-    Ok(())
 }
 
-fn set_panic_hook() {
-    match *LOG_FORMAT {
+fn set_panic_hook(opts: &PrismaOpt) {
```

Comment:
Why not passing the `LogFormat` directly?

Reply:
Oh sure thing!